rs/system_api/src/lib.rs

```diff
@@ -1482,7 +1491,50 @@ impl SystemApiImpl {
     }
 
     pub fn into_system_state_changes(self) -> SystemStateChanges {
-        self.sandbox_safe_system_state.system_state_changes
```

Comment:
Is this change supposed to prevent accidental changes to the system state from being committed? In other words, why are we changing also other cases beyond replicated queries?

Reply:
Yeah, I wanted to make it more clear what changes are reported from the system api. It depends on the context and I felt not returning the ones that are not relevant is better (especially given that other layers will now expect to apply whatever they get from `SystemStateChanges`).

Reply:
That's a good point, let me do it.